common/service.go

```diff
@@ -139,17 +146,17 @@ func (bs *BaseService) Stop() bool {
 func (bs *BaseService) OnStop() {}
 
 // Implements Service
-func (bs *BaseService) Reset() (bool, error) {
+func (bs *BaseService) Reset() error {
 	if !atomic.CompareAndSwapUint32(&bs.stopped, 1, 0) {
 		bs.Logger.Debug(Fmt("Can't reset %v. Not stopped", bs.name), "impl", bs.impl)
-		return false, nil
+		return fmt.Errorf("can't reset running %s", bs.name)
```

Comment:
maybe we should `panic` here. like clearly if we are trying to reset running service, then something wrong with our code?

Reply:
We could, but it's probably best if we try to minimize panics in libraries ?